File: python/first_non_repeating.py
def first_non_repeating(s: str) -> str | None:
    count_map = {}

    for c in s:
        count_map[c] = count_map.get(c, 0) + 1

    for c in s:
        if count_map[c] == 1:
            return c

    return None


if __name__ == "__main__":
    test_cases = ["swiss", "aabbcc", "abcabc", "leetcode"]

    for s in test_cases:
        result = first_non_repeating(s)
        if result:
            print(f'First non-repeating character in "{s}": {result}')
        else:
            print(f'No non-repeating character found in "{s}"')

File: csharp/src/FirstNonRepeating.cs
using System;
using System.Collections.Generic;

class FirstNonRepeating
{
    static char? FindFirstNonRepeating(string s)
    {
        var countMap = new Dictionary<char, int>();

        foreach (char c in s)
        {
            countMap[c] = countMap.GetValueOrDefault(c, 0) + 1;
        }

        foreach (char c in s)
        {
            if (countMap[c] == 1)
                return c;
        }

        return null;
    }

    static void Main(string[] args)
    {
        string[] testCases = { "swiss", "aabbcc", "abcabc", "leetcode" };

        foreach (string s in testCases)
        {
            char? result = FindFirstNonRepeating(s);
            if (result.HasValue)
                Console.WriteLine($"First non-repeating character in \"{s}\": {result}");
            else
                Console.WriteLine($"No non-repeating character found in \"{s}\"");
        }
    }
}
